optaplanner-examples/src/main/java/org/optaplanner/examples/examination/solver/score/ExaminationConstraintProvider.java

```diff
@@ -99,8 +115,8 @@ protected Constraint periodPenaltyExclusion(ConstraintFactory constraintFactory)
                         filtering((periodPenalty, leftExam) -> leftExam.getPeriod() != null))
                 .join(Exam.class,
                         equal((periodPenalty, leftExam) -> periodPenalty.getRightTopic(), Exam::getTopic),
-                        filtering((periodPenalty, leftExam, rightExam) -> rightExam.getPeriod() != null),
-                        filtering((periodPenalty, leftExam, rightExam) -> leftExam.getPeriod().equals(rightExam.getPeriod())))
+                        equal((periodPenalty, leftExam) -> leftExam.getPeriod(), Exam::getPeriod),
+                        filtering((periodPenalty, leftExam, rightExam) -> rightExam.getPeriod() != null))
```

Comment:
2 lines above says that the leftExam's period isn't null.
The line above says that the rightExam period is the same one.
Therefore, checking if the rightExam isn't null is obsolete.
If A != null && A == B then B != null